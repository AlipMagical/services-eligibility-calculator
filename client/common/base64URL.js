/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const base64URLEncode = plainText =>
  btoa(plainText)
    .replace(/=/g, '')
    .replace(/\+/g, '-')
    .replace(/\//g, '_');

export const base64URLDecode = encodedText =>
  atob(encodedText.replace(/-/g, '+').replace(/_/g, '/'));

export default { base64URLEncode, base64URLDecode };
