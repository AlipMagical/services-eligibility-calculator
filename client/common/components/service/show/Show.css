/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.accepting-applications,
.not-accepting-applications,
.accepting-applications-future {
  display: inline-block;
  width: 100%;
  text-align: center;
  margin-top: 24px;
  border-radius: 8px;
}
.accepting-applications {
  color: #1e8e3e;
  background: #e6f4ea;
  border: 1px solid #a8dab5;
}
.not-accepting-applications {
  color: #5f6368;
}
.accepting-applications-future {
  background: #f1f3f4;
  color: #5f6368;
}

.apply-icon {
  margin-right: 20px;
}

.service-title {
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 24px;
}

.material-icons {
  font-size: 20px;
}

.eligibility {
  align-items: center;
}

.eligibility .material-icons {
  padding: 5px;
  border-radius: 20px;
  color: #80868b;
  background: #f1f3f4;
}

.eligibility .col {
  margin-left: 16px;
}

.eligibility-other {
  margin-top: 7px;
}

.eligibility-or {
  text-align: center;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
  line-height: 0;
  margin-bottom: 32px;
}

.eligibility-or span {
  background: white;
  padding: 0 10px;
}

.max-income .title {
  margin-top: 0px;
  margin-bottom: 15px;
  font-size: 32px;
  line-height: 40px;
}

.max-income .row {
  margin: 0px;
}

.max-income .col {
  padding: 0px;
}
.max-income hr {
  margin: 8px;
}
