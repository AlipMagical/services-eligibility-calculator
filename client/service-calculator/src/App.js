/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import React, { createContext, useState } from 'react';

import { ANALYTICS_ID } from './common/config';
import { Body } from './components/Body';
import { DataContext } from './DataContext';
import Footer from './components/Footer';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import { Router } from 'react-router-dom';
import ScrollToTop from './common/utils/ScrollToTop';
import { createBrowserHistory } from 'history';

export const FilterContext = createContext();

export const createInitialFilterState = () => ({
  income: undefined,
  interests: {},
  people: {}
});

const history = createBrowserHistory();

export default function App(props) {
  const [analyticsInitialized, setAnalyticsInitialized] = useState(false);
  const [lastSearch, setLastSearch] = useState('');
  const [filterState, setFilterState] = useState(createInitialFilterState());
  const resetFilterState = () => setFilterState(createInitialFilterState());

  if (!analyticsInitialized) {
    ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });
    setAnalyticsInitialized(true);
  }

  return (
    <DataContext.StateProvider>
      <FilterContext.Provider
        value={{
          filterState,
          setFilterState,
          resetFilterState,
          lastSearch,
          setLastSearch
        }}
      >
        <Router history={history}>
          <ScrollToTop />
          <Body />
          <Footer />
        </Router>
      </FilterContext.Provider>
    </DataContext.StateProvider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
