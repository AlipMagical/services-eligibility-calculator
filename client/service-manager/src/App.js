/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import React, { createContext, useState } from 'react';
import {
  getLoggedInUser,
  getMillisUntilExpiration,
  isLoggedIn
} from './common/auth';

import { ANALYTICS_ID } from './common/config';
import { BASEPATH } from './common/config';
import Header from './components/Header';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import Toast from './components/Toast';
import departmentRoutes from './routes/department';
import getUserConfirmation from './getUserConfirmation';
import interestRoutes from './routes/interest';
import serviceRoutes from './routes/service';
import settingsRoutes from './routes/settings';
import { useSettings } from './hooks';

export const AuthContext = createContext();
export const ToastContext = createContext();
export const SettingsContext = createContext();

export default function App(props) {
  ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });

  const [loginState, setLoginState] = useState({
    isLoggedIn: isLoggedIn(),
    loggedInUser: getLoggedInUser()
  });

  const onLoginStatusChanged = () => {
    //setTimeout to update when token might expire.
    const millis = getMillisUntilExpiration();
    millis && setTimeout(onLoginStatusChanged, millis);

    setLoginState({
      isLoggedIn: isLoggedIn(),
      loggedInUser: getLoggedInUser()
    });
  };

  const { settings, setSettings } = useSettings();

  const [toastMessage, setToastMessage] = useState(null);

  return (
    <AuthContext.Provider value={{ ...loginState, onLoginStatusChanged }}>
      <SettingsContext.Provider value={{ settings, setSettings }}>
        <ToastContext.Provider value={{ toastMessage, setToastMessage }}>
          <BrowserRouter getUserConfirmation={getUserConfirmation}>
            <Header />
            <div className="container bd-content">
              <Switch>
                {serviceRoutes}
                {settingsRoutes}
                {interestRoutes}
                {departmentRoutes}
                <Redirect from={BASEPATH} exact to={BASEPATH + 'services/'} />
                <Route render={() => <h1>Not Found</h1>} />
              </Switch>
            </div>
            {toastMessage && <Toast message={toastMessage} />}
          </BrowserRouter>
        </ToastContext.Provider>
      </SettingsContext.Provider>
    </AuthContext.Provider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
