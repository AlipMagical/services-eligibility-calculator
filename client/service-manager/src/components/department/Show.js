/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState } from 'react';

import { Modal } from 'react-bootstrap';
import PropTypes from 'prop-types';
import Remove from './Remove';
import { base64URLDecode } from '../../common/base64URL';
import { defaultLocale } from '../../Locales';
import { renderNonDefaultLocales } from '../namedTranslatedEntity';
import { useDepartmentTranslations } from '../../hooks';

export default function Show(props) {
  const departmentTranslations = useDepartmentTranslations(
    base64URLDecode(props.departmentId)
  );
  const [showRemoveModal, setShowRemoveModal] = useState(false);
  const department = departmentTranslations[defaultLocale];

  return (
    <>
      {showRemoveModal && (
        <Remove
          departmentId={props.departmentId}
          onCancel={() => {
            setShowRemoveModal(false);
          }}
          onRemove={() => {
            // Unwind the stack of modals.
            setShowRemoveModal(false);
            props.onCancel();
          }}
        />
      )}
      <Modal size="xl" show={showRemoveModal === false} onHide={props.onCancel}>
        <Modal.Body>
          <h1>Department: {department && department['name']}</h1>
          <>
            {department === undefined && (
              <div className="alert alert-info" role="status">
                Loading...
              </div>
            )}

            {department && (
              <table className="table table-responsive">
                <thead>
                  <tr>
                    <th>Field</th>
                    <th>Value</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th scope="row">name</th>
                    <td>
                      {department['name']}{' '}
                      {renderNonDefaultLocales('name', departmentTranslations)}
                    </td>
                  </tr>
                </tbody>
              </table>
            )}
            <div className="float-left">
              <button onClick={props.onCancel} className="btn btn-blue-text">
                Close
              </button>
            </div>
            <div className="float-right">
              <button
                onClick={() => setShowRemoveModal(true)}
                className="btn btn-blue-text"
              >
                Remove
              </button>
              {
                <button onClick={props.onEdit} className="btn btn-blue-text">
                  Edit
                </button>
              }
            </div>
          </>
        </Modal.Body>
      </Modal>
    </>
  );
}

Show.propTypes = {
  departmentId: PropTypes.string,
  onCancel: PropTypes.func,
  onEdit: PropTypes.func
};
