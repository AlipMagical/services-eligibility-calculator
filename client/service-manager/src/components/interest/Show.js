/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState } from 'react';

import { Modal } from 'react-bootstrap';
import PropTypes from 'prop-types';
import Remove from './Remove';
import { base64URLDecode } from '../../common/base64URL';
import { defaultLocale } from '../../Locales';
import { renderNonDefaultLocales } from '../namedTranslatedEntity';
import { useInterestTranslations } from '../../hooks';

export default function Show(props) {
  const interestTranslations = useInterestTranslations(
    base64URLDecode(props.interestId)
  );
  const [showRemoveModal, setShowRemoveModal] = useState(false);
  const interest = interestTranslations[defaultLocale];

  return (
    <>
      {showRemoveModal && (
        <Remove
          interestId={props.interestId}
          onCancel={() => {
            setShowRemoveModal(false);
          }}
          onRemove={() => {
            // Unwind the stack of modals.
            setShowRemoveModal(false);
            props.onCancel();
          }}
        />
      )}
      <Modal size="xl" show={showRemoveModal === false} onHide={props.onCancel}>
        <Modal.Body>
          <h1>Interest: {interest && interest['name']}</h1>
          <>
            {interest === undefined && (
              <div className="alert alert-info" role="status">
                Loading...
              </div>
            )}
            {interest && (
              <table className="table table-responsive">
                <thead>
                  <tr>
                    <th>Field</th>
                    <th>Value</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th scope="row">name</th>
                    <td>
                      {interest['name']}{' '}
                      {renderNonDefaultLocales('name', interestTranslations)}
                    </td>
                  </tr>
                  <tr>
                    <th scope="row">materialIcon</th>
                    <td>
                      <div className="row">
                        <div className="col-1">
                          <i className="material-icons">
                            {interest['materialIcon']}
                          </i>
                        </div>
                        <div className="col">{interest['materialIcon']}</div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            )}
            <div className="float-left">
              <button onClick={props.onCancel} className="btn btn-blue-text">
                Close
              </button>
            </div>
            <div className="float-right">
              <button
                onClick={() => setShowRemoveModal(true)}
                className="btn btn-blue-text"
              >
                Remove
              </button>
              {
                <button onClick={props.onEdit} className="btn btn-blue-text">
                  Edit
                </button>
              }
            </div>
          </>
        </Modal.Body>
      </Modal>
    </>
  );
}

Show.propTypes = {
  interestId: PropTypes.string,
  onCancel: PropTypes.func,
  onEdit: PropTypes.func
};
