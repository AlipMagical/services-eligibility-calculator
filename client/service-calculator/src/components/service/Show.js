/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import '../../common/components/service/show/Show.css';

import { Col, Row } from 'react-bootstrap';
import { ExternalLink, sendEmail } from '../../common/utils';
import React, { useContext } from 'react';

import ApplicationCard from '../../common/components/service/show/ApplicationCard';
import { BASEPATH } from '../../common/config';
import { DataContext } from '../../DataContext';
import DetailsCard from '../../common/components/service/show/DetailsCard';
import EligibilityCard from '../../common/components/service/show/EligibilityCard';
import { FilterContext } from '../../App';
import Helmet from 'react-helmet';
import OverviewCard from '../../common/components/service/show/OverviewCard';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import { TransparentButton } from '../eligibility-calculator/TransparentButton';
import { base64URLDecode } from '../../common/base64URL';
import { getTotalPeople } from '../../filters/filterAge';
import { useTranslation } from 'react-i18next';

export default function Show(props) {
  const serviceId = base64URLDecode(props.match.params.id);
  const { settings, servicesById } = useContext(DataContext);
  const service = servicesById && servicesById[serviceId];
  const { t } = useTranslation();
  const { filterState, lastSearch } = useContext(FilterContext);

  const shareViaEmail = () => {
    var emailBody = t('serviceDetails.shareViaEmailMessage');
    emailBody += '%0D%0A%0D%0A';
    emailBody += ((service && service.name) || '') + '%0D%0A';
    emailBody += encodeURIComponent(window.location.href);

    var emailSubject = t('serviceDetails.shareViaEmailSubject');
    sendEmail('', emailSubject, emailBody);

    ReactGA.event({
      category: 'Click',
      action: 'Share',
      label: 'Shared by Email'
    });
  };

  let resultsPage = BASEPATH;
  if (lastSearch) {
    resultsPage += '?q=' + encodeURIComponent(lastSearch);
  }

  return (
    <>
      {service === undefined && (
        <div role="status">
          <h2>{t('loading')}</h2>
        </div>
      )}

      {service && (
        <div className="service-details mx-auto">
          <Helmet>
            <title>{service.name}</title>
          </Helmet>
          <Row>
            <Col xs={12} className="mt-3">
              <TransparentButton
                className="d-xl-none"
                onClick={() => props.history.push(resultsPage)}
              >
                <i className="material-icons align-top" aria-hidden="true">
                  arrow_back
                </i>{' '}
                {t('serviceDetails.backButton')}
              </TransparentButton>
              <h1
                id="service-details-name"
                className="mt-1 mb-3 service"
                tabIndex="-1"
              >
                {service.name}
              </h1>
            </Col>
          </Row>
          <Row className="gutter-24">
            <Col xs={12} className="px-0 px-md-4">
              <OverviewCard service={service} hideTitle>
                <Row className="mt-4">
                  <Col xs={6}>
                    <button
                      className="btn btn-share w-100"
                      onClick={() => {
                        if (navigator.share) {
                          navigator
                            .share({
                              title: service.name,
                              text: service.name,
                              url: window.location.href
                            })
                            .then(() => {
                              ReactGA.event({
                                category: 'Click',
                                action: 'Share',
                                label: 'Native Share'
                              });
                            })
                            .catch(e => {
                              if (e.name === 'AbortError') {
                                // User cancelled share.
                                // Don't display sharing modal.
                                ReactGA.event({
                                  category: 'Click',
                                  action: 'Share',
                                  label: 'Cancelled'
                                });
                                return;
                              }
                              shareViaEmail();
                            });
                        } else {
                          shareViaEmail();
                        }
                      }}
                    >
                      {t('serviceDetails.share')}
                    </button>
                  </Col>
                  <Col xs={6}>
                    <ExternalLink
                      action="LearnMore"
                      url={service.applicationOnline}
                      className="btn btn-primary w-100"
                    >
                      {t('serviceDetails.learnMore')}
                    </ExternalLink>
                  </Col>
                </Row>
              </OverviewCard>
              <DetailsCard service={service} />
              <EligibilityCard
                eligibilityProfiles={service.eligibilityProfiles}
                ami={settings && settings.ami}
                fpl={settings && settings.fpl}
                householdSize={getTotalPeople(filterState.people)}
              />
              <ApplicationCard service={service} />
            </Col>
          </Row>
        </div>
      )}
    </>
  );
}

Show.propTypes = {
  match: PropTypes.object,
  onEdit: PropTypes.func
};
