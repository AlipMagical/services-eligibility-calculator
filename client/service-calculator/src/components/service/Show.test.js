/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FilterContext, createInitialFilterState } from '../../App';
import { MemoryRouter, Route } from 'react-router-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import React from 'react';
import Show from './Show';
import { act } from 'react-dom/test-utils';
import { base64URLEncode } from '../../common/base64URL';
import { render } from 'react-dom';
import { DataContext } from '../../DataContext';

expect.extend(toHaveNoViolations);

const filterState = createInitialFilterState();

it('renders show a service modal', async () => {
  const container = document.createElement('div');

  fetch.setupDataContextFetches();

  await act(async () => {
    return render(
      <MemoryRouter>
        <DataContext.StateProvider>
          <FilterContext.Provider value={{ filterState }}>
            <Route
              component={props => (
                <Show
                  {...props}
                  match={{
                    params: {
                      id: base64URLEncode('/api/services/0')
                    }
                  }}
                />
              )}
            />
          </FilterContext.Provider>
        </DataContext.StateProvider>
      </MemoryRouter>,
      container
    );
  });

  expect(container).toMatchSnapshot();
  expect(await axe(container)).toHaveNoViolations();
});
