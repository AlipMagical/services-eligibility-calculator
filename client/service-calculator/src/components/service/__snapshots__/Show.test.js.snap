// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders show a service modal 1`] = `
<div>
  <div
    class="service-details mx-auto"
  >
    <div
      class="row"
    >
      <div
        class="mt-3 col-12"
      >
        <button
          class="btn btn-clear btn-gutter d-xl-none"
        >
          <i
            aria-hidden="true"
            class="material-icons align-top"
          >
            arrow_back
          </i>
           
          Back
        </button>
        <h1
          class="mt-1 mb-3 service"
          id="service-details-name"
          tabindex="-1"
        >
          Accusantium aperiam consequatur.
        </h1>
      </div>
    </div>
    <div
      class="gutter-24 row"
    >
      <div
        class="px-0 px-md-4 col-12"
      >
        <div
          class="service-card card"
          tabindex="-1"
        >
          <div
            class="card-body"
          >
            <div
              style="font-size: 16px;"
            >
              <div>
                Rem reprehenderit delectus architecto vero numquam aut repellendus.
              </div>
            </div>
            <div
              class="accepting-applications-future inline box"
            >
              <div>
                <i
                  aria-hidden="true"
                  class="material-icons"
                >
                  schedule
                </i>
              </div>
              <div>
                Accepting applications starting 
                Apr 27, 2020
              </div>
            </div>
            <div
              class="mt-4 row"
            >
              <div
                class="col-6"
              >
                <button
                  class="btn btn-share w-100"
                >
                  Share
                </button>
              </div>
              <div
                class="col-6"
              >
                <a
                  class="btn btn-primary w-100"
                  href="https://marquardt.biz/possimus-cupiditate-iure-quo-sit-ratione.html"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn More
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="service-card card"
          tabindex="-1"
        >
          <div
            class="card-header"
          >
            What is the service?
            <button
              aria-label="minimize"
              class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                remove
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="card-body"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div>
                  Impedit non assumenda similique. Voluptates sunt tempora voluptas accusantium alias reprehenderit veritatis. Eos doloribus et magnam est enim totam necessitatibus.
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                Department
                :
              </div>
              <div
                class="text-break col"
              >
                Housing and Human Services
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                Application window
                :
              </div>
              <div
                class="text-break col"
              >
                Apr 27, 2020
                 - 
                Sep 3, 2020
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                Relates to
                :
              </div>
              <div
                class="text-break col"
              >
                <span>
                  Wellness
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="service-card card"
          tabindex="-1"
        >
          <div
            class="card-header"
          >
            Who is eligible?
            <button
              aria-label="minimize"
              class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                remove
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="card-body"
          />
        </div>
        <div
          class="service-card card"
          tabindex="-1"
        >
          <div
            class="card-header"
          >
            How do I apply?
            <button
              aria-label="minimize"
              class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                remove
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="card-body"
          >
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                Learn more
                :
              </div>
              <div
                class="col"
              >
                <a
                  href="https://marquardt.biz/possimus-cupiditate-iure-quo-sit-ratione.html"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  https://marquardt.biz/possimus-cupiditate-iure-quo-sit-ratione.html
                </a>
              </div>
            </div>
            <hr />
            <div
              class="instructions"
            >
              First, you'll need
              :
            </div>
            <div
              class="row"
            >
              <div
                class="text-break col"
              >
                <div>
                  Totam porro consequatur molestias recusandae non hic. Et libero aperiam commodi dolor.
                </div>
              </div>
            </div>
            <hr />
            <div
              class="instructions"
            >
              When you're ready, apply
              :
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                <i
                  aria-hidden="true"
                  class="material-icons apply-icon align-middle"
                >
                  phone
                </i>
                <span
                  class="align-middle"
                >
                  Call:
                </span>
              </div>
              <div
                class="col"
              >
                <a
                  href="tel:827.937.7299"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  827.937.7299
                </a>
                 
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                <i
                  aria-hidden="true"
                  class="material-icons apply-icon align-middle"
                >
                  email
                </i>
                <span
                  class="align-middle"
                >
                  Email:
                </span>
              </div>
              <div
                class="col"
              >
                <a
                  href="mailto:ebert.harley@prosacco.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  ebert.harley@prosacco.com
                </a>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                <i
                  aria-hidden="true"
                  class="material-icons apply-icon align-middle"
                >
                  computer
                </i>
                <span
                  class="align-middle"
                >
                  Online:
                </span>
              </div>
              <div
                class="col"
              >
                <a
                  href="http://www.oconner.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  http://www.oconner.com/
                </a>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="label col-md-4"
              >
                <i
                  aria-hidden="true"
                  class="material-icons apply-icon align-middle"
                >
                  directions_walk
                </i>
                <span
                  class="align-middle"
                >
                  In person:
                </span>
              </div>
              <div
                class="col"
              >
                <a
                  href="https://www.google.com/maps/search/?api=1&query=4070%20Torphy%20Isle%2C%20Stokesburgh%2C%20VT%2032336"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div>
                    4070 Torphy Isle, Stokesburgh, VT 32336
                  </div>
                </a>
                <br />
                <div>
                  A eaque tempora quia ex id qui assumenda.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
